src/i18n/i18nConfig.ts

```diff
@@ -3,6 +3,30 @@ import { initReactI18next } from "react-i18next";
 import LanguageDetector from "i18next-browser-languagedetector";
 import en from "./locales/en.json";
 import hi from "./locales/hi.json";
+import api from "~/common/lib/api";
+
+export const settings = {
```

Comment:
According to https://github.com/i18next/i18next-browser-languageDetector/blob/master/index.d.ts#L75 `addDetector` is [taking `dectector` (`settings`) as `CustomDetector`](https://github.com/i18next/i18next-browser-languageDetector/blob/master/index.d.ts#L64-L68).

So tried:

```ts
import type CustomDetector from "i18next-browser-languagedetector";

export const settings: CustomDetector = {
  name: "settings",

  cacheUserLanguage(lng, options) {
    // options -> are passed in options
    // lng -> current language, will be called after init and on changeLanguage
    // store it
    api.getSettings().then((setting) => {
      i18n.changeLanguage(setting.locale);
    });
  },

  lookup(options) {
    // options -> are passed in options
    api.getSettings().then((setting) => {
      i18n.changeLanguage(setting.locale);
    });
  },
};
```

Which then gives me:
![image](https://user-images.githubusercontent.com/1016218/169205381-43003e0a-b82f-4247-b0ac-65b10c242fdf.png)

Ugh. Looks like it's not using the right type or not importing the right one?
Ideas @dylancom ?

Reply:
Looks like `import type { CustomDetector } from "i18next-browser-languagedetector";` works better (bit confused about this tbh). Now `lookup` needs to return `string | string[] | undefined`. 

![image](https://user-images.githubusercontent.com/1016218/169205983-8cb0e5e2-52b8-468b-9edc-9611f525d059.png)

Not sure what lookup really needs to return here but maybe that helps. This would be nice to pair ;)

Reply:
Ok sounds good, we can pair on this next week to finish it off. :)

Reply:
Let's do it! It's important to understand how to tackle these things.